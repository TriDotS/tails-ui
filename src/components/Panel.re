let component = ReasonReact.statelessComponent("Panel");

let make = (~color, ~title, ~extra, ~children) => {
  ...component,
  render: _self =>
    <div className={j|border-2 border-$(color)-light rounded|j}>
      <div
        className={j|flex flex-row items-center justify-between w-full border-b-2 border-$(color)-light p-2|j}>
        <div className="text-2xl text-grey-darker">
          (ReasonReact.stringToElement(title))
        </div>
        <div> (ReasonReact.arrayToElement(extra)) </div>
      </div>
      <div className="p-2"> (ReasonReact.arrayToElement(children)) </div>
    </div>
};

let default =
  ReasonReact.wrapReasonForJs(~component, jsProps =>
    make(
      ~color=jsProps##color,
      ~title=jsProps##title,
      ~extra=jsProps##extra,
      ~children=jsProps##children
    )
  );
