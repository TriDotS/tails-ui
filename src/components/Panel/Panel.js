import PropTypes from 'prop-types'
import React from 'react'

const Panel = ({ color, title, extra, children }) => (
  <div className={`border-2 border-${color}-light rounded`}>
    <div
      className={`flex flex-row items-center justify-between w-full border-b-2 border-${color}-light p-2 `}
    >
      <div className="text-2xl">{title}</div>
      <div>{extra}</div>
    </div>
    <div className="p-2">{children}</div>
  </div>
)

Panel.defaultProps = {
  color: 'grey'
}

Panel.propTypes = {
  color: PropTypes.string,
  title: PropTypes.string.isRequired,
  extra: PropTypes.node,
  children: PropTypes.node.isRequired
}

export default Panel
