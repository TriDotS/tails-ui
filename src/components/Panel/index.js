import PropTypes from 'prop-types'
import React from 'react'

const Panel = ({ color, title, extra, children }) => (
  <div className={`border-2 border-${color}-light rounded`}>
    <div
      className={`flex flex-row items-center justify-between w-full border-b-2 border-${color}-light p-4 `}
    >
      <div className="text-2xl text-grey-darker">{title}</div>
      <div>{extra}</div>
    </div>
    <div className="p-4">{children}</div>
  </div>
)

Panel.defaultProps = {
  color: 'grey'
}

Panel.propTypes = {
  color: PropTypes.string,
  title: PropTypes.string.isRequired,
  extra: PropTypes.node,
  children: PropTypes.node.isRequired
}

export default Panel
