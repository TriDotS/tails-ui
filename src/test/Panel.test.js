import Panel from '../components/Panel/Panel'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<Panel title="A Panel">A panel</Panel>, div)
})

test('panel snapshot', () => {
  const tree = renderer.create(<Panel title="A Panel">A panel</Panel>).toJSON()
  expect(tree).toMatchSnapshot()
})
