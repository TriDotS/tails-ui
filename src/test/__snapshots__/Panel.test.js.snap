// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panel snapshot 1`] = `
<div
  className="border-2 border-grey-light rounded"
>
  <div
    className="flex flex-row items-center justify-between w-full border-b-2 border-grey-light p-4 "
  >
    <div
      className="text-2xl text-grey-darker"
    >
      A Panel
    </div>
    <div />
  </div>
  <div
    className="p-4"
  >
    A panel
  </div>
</div>
`;
